plugins/PrivacyManager/PrivacyManager.php

```diff
@@ -582,4 +586,22 @@ private static function getMaxGoalId()
     {
         return Db::fetchOne("SELECT MAX(idgoal) FROM " . Common::prefixTable('goal'));
     }
+
+    /**
+     * Returns a unique salt used for pseudonimisation of user id only
+     *
+     * @return string
+     */
+    public static function getUserIdSalt()
+    {
+        static $salt = null;
+        if (is_null($salt)) {
+            $salt = Option::get(self::OPTION_USERID_SALT);
+            if (empty($salt)) {
+                $salt = Common::generateUniqId();
```

Comment:
You could generate a stronger salt like `Common::getRandomString($len = 40, $alphabet = "abcdefghijklmnoprstuvwxyzABCDEFGHIJKLMNOPRSTUVWXYZ0123456789_-$");`
Also you may want to write an update method for this as it be better to avoid DB connections during tracker. You would also need to add a test to see if it actually works during tracker mode... because it might not actually work when operating in tracker mode. I had this a few times as some methods like `fetchOne` may not be available during tracker mode.